ni_measurementlink_service/measurement/info.py

```diff
@@ -67,26 +67,107 @@ class TypeSpecialization(enum.Enum):
-    PinArray1D = (type_pb2.Field.TYPE_STRING, True, TypeSpecialization.Pin)
-    PathArray1D = (type_pb2.Field.TYPE_STRING, True, TypeSpecialization.Path)
-    EnumArray1D = (type_pb2.Field.TYPE_ENUM, True, TypeSpecialization.Enum)
+    Int32 = 0
+    Int64 = 1
+    UInt32 = 2
+    UInt64 = 3
+    Float = 4
+    Double = 5
+    Boolean = 6
+    String = 7
+    Pin = 8
+    Path = 9
+    Enum = 10
+
+    Int32Array1D = 100
+    Int64Array1D = 101
+    UInt32Array1D = 102
+    UInt64Array1D = 103
+    FloatArray1D = 104
+    DoubleArray1D = 105
+    BooleanArray1D = 106
+    StringArray1D = 107
+    PinArray1D = 108
+    PathArray1D = 109
+    EnumArray1D = 110
+
+
+class DataTypeInfo(NamedTuple):
+    """Class that represents the information for each of the DataType enum values.
+
+    Attributes
+    ----------
+        grpc_field_type (type_pb2.Field.Kind): Field.Kind associated with the DataType
+
+        repeated (bool): Whether the DataType is a repeated field
+
+        type_specialization (TypeSpecialization): Specific type when value_type
+        can have more than one use
+
+    """
+
+    grpc_field_type: type_pb2.Field.Kind.ValueType
+    repeated: bool
+    type_specialization: TypeSpecialization = TypeSpecialization.NoType
+
+
+Int32TypeInfo = DataTypeInfo(type_pb2.Field.TYPE_INT32, False)
+Int64TypeInfo = DataTypeInfo(type_pb2.Field.TYPE_INT64, False)
+UInt32TypeInfo = DataTypeInfo(type_pb2.Field.TYPE_UINT32, False)
+UInt64TypeInfo = DataTypeInfo(type_pb2.Field.TYPE_UINT64, False)
+FloatTypeInfo = DataTypeInfo(type_pb2.Field.TYPE_FLOAT, False)
+DoubleTypeInfo = DataTypeInfo(type_pb2.Field.TYPE_DOUBLE, False)
+BooleanTypeInfo = DataTypeInfo(type_pb2.Field.TYPE_BOOL, False)
+StringTypeInfo = DataTypeInfo(type_pb2.Field.TYPE_STRING, False)
+PinTypeInfo = DataTypeInfo(type_pb2.Field.TYPE_STRING, False, TypeSpecialization.Pin)
+PathTypeInfo = DataTypeInfo(type_pb2.Field.TYPE_STRING, False, TypeSpecialization.Path)
+EnumTypeInfo = DataTypeInfo(type_pb2.Field.TYPE_ENUM, False, TypeSpecialization.Enum)
+
+Int32Array1DTypeInfo = DataTypeInfo(type_pb2.Field.TYPE_INT32, True)
+Int64Array1DTypeInfo = DataTypeInfo(type_pb2.Field.TYPE_INT64, True)
+UInt32Array1DTypeInfo = DataTypeInfo(type_pb2.Field.TYPE_UINT32, True)
+UInt64Array1DTypeInfo = DataTypeInfo(type_pb2.Field.TYPE_UINT64, True)
+FloatArray1DTypeInfo = DataTypeInfo(type_pb2.Field.TYPE_FLOAT, True)
+DoubleArray1DTypeInfo = DataTypeInfo(type_pb2.Field.TYPE_DOUBLE, True)
+BooleanArray1DTypeInfo = DataTypeInfo(type_pb2.Field.TYPE_BOOL, True)
+StringArray1DTypeInfo = DataTypeInfo(type_pb2.Field.TYPE_STRING, True)
+PinArray1DTypeInfo = DataTypeInfo(type_pb2.Field.TYPE_STRING, True, TypeSpecialization.Pin)
+PathArray1DTypeInfo = DataTypeInfo(type_pb2.Field.TYPE_STRING, True, TypeSpecialization.Path)
+EnumArray1DTypeInfo = DataTypeInfo(type_pb2.Field.TYPE_ENUM, True, TypeSpecialization.Enum)
+
+
+class DataTypeInfoLookup:
```

Comment:
I don't think it's necessary for the lookup to have a separate class.

Options:
- Define the lookup function at file scope
- Define the lookup function as a static method of the DataTypeInfo class